src/components/Todo.tsx

```diff
@@ -3,17 +3,16 @@ import { deleteTodo } from "../actions/index";
 import cross from "../assets/closee.png";
 import { useDispatch } from "react-redux";
 import { markTodo, unmarkTodo, changeTodo } from "../actions/index";
+import { ListItem } from "../utils/models";
 
 type TodoProps = {
-  list: {
-    data: string;
-    id: number;
-    completed: boolean;
-  };
+  list: ListItem;
 };
 
 const Todo = memo((props: TodoProps) => {
   const { list } = props;
+  const dispatch = useDispatch();
+
   const [toggle, setToggle] = useState(true);
   const [text, setText] = useState(list.data);
```

Comment:
Add type to the state eg. `useState<stirng>(...)`. Also, `list.data` can lead to errors. Try to provide a fallback value.